### PR TITLE
[BACKLOG-24021] Add functionality for handling deprecated PDI job ent…

### PR DESCRIPTION
**Pull request by ddiroma**
…ries in a consistent manner

### CODE REVIEW

File: engine/src/main/java/org/pentaho/di/job/JobPainter.java [hunk 1]
```diff
@@ -241,7 +241,13 @@ protected void drawJobEntryCopy( JobEntryCopy jobEntryCopy ) {
           .add( new AreaOwner( AreaType.JOB_ENTRY_ICON, x, y, iconsize, iconsize, offset, subject, jobEntryCopy ) );
     }
 
-    gc.setForeground( jobEntryCopy.isMissing() ? EColor.RED : EColor.CRYSTAL );
+    if ( jobEntryCopy.isMissing() ) {
+      gc.setForeground( EColor.RED );
+    } else if ( jobEntryCopy.isDeprecated() ) {
+      gc.setForeground( 246, 196, 56 );
```
Comment (pentaho-amessier) [inline engine/src/main/java/org/pentaho/di/job/JobPainter.java]: Maybe this color could be a constant?

**Comment by ddiroma** [comment]
@pentaho/r2d2 

Wingman was down yesterday and is backed up this morning. Please review if you want.

File: ui/src/main/resources/org/pentaho/di/ui/spoon/job/messages/messages_en_US.properties [hunk 2]
```diff
@@ -247,3 +247,9 @@ JobGraph.Dialog.LoopAfterHopEnabled.Message=Enabling this hop creates a loop tha
 JobGraph.Dialog.HopFlipCausesLoop.Message=Flipping the direction of this hop creates a loop that might cause errors when this job runs.
 JobGraph.Dialog.HopCausesLoop.Message=Adding this hop creates a loop that might cause errors when this job runs.
 JobGraph.Dialog.HopCausesLoop.Title=Warning
+
+JobGraph.DeprecatedEntry.Tooltip.Title=This is a deprecated step and will be removed.
+JobGraph.DeprecatedEntry.Tooltip.Message1={0} will be removed in an upcoming
```
Comment (pentaho-amessier) [inline ui/src/main/resources/org/pentaho/di/ui/spoon/job/messages/messages_en_US.properties]: Out of curiosity, is it not possible to provide this text as a single message with new line characters in appropriate places?

**Comment by ddiroma** [comment]
@pentaho-amessier made the desired changes

**Pull request merged**